### PR TITLE
setup services refactor 

### PR DESCRIPTION
**Pull request by almogdepaz**
tech debt regarding creation of services, 
will help in type hinting and  eventually taking setup_nodes.py and setup_services.py out of mypy ignore list 

currently we return multiple return values on service creation 

this will change so we always return the service and extract needed objects in the caller

### CODE REVIEW

**Review by altendky — DISMISSED** [review]
thanks

**Comment by emlowe** [comment]
Close and reopen for NFT wallet tests

**Review by emlowe — APPROVED** [review]

**Comment by altendky** [comment]
The link for the supposedly failed CodeQL job just 500s for me and there's no record of a fail in the action history so just closing and reopening to trigger a rerun.

File: chia/simulator/setup_services.py [hunk 1]
```diff
@@ -137,11 +136,7 @@ async def setup_full_node(
         )
     await service.start()
 
-    # TODO, just always yield the service only and adjust all other places
-    if yield_service:
-        yield service
-    else:
-        yield service._api
+    yield service
```
Comment (aqk) [inline chia/simulator/setup_services.py:139]: So much nicer!

**Pull request merged**